lib/hard-module.js

```diff
@@ -56,7 +61,7 @@ function needRebuild(buildTimestamp, fileDependencies, contextDependencies, file
     if(!ts) timestamp = Infinity;
     if(ts > timestamp) timestamp = ts;
   });
-  return timestamp >= buildTimestamp;
+  return timestamp >= buildTimestamp && needsMd5Rebuild;
 }
 
 HardModule.needRebuild = needRebuild;
```

Comment:
do i need to be passing in the md5 information to the constructor for the instance method to also check md5? is this only used in watch mode?